### ISSUE TITLE
Ensure that all database-related streams are closed

### ISSUE THREAD

**Issue by arteymix** [PavlidisLab/rdp#264]
Review any code in the repository layer that involves usage of stream. Those should be closed right away in the service layer that consumes them.

Reference: https://docs.jboss.org/hibernate/orm/5.3/javadocs/org/hibernate/query/Query.html#stream()

**Comment by arteymix** [comment]
I will also do a check in Gemma to see if we can run into this issue.

I was doing some work on optimizing the search for GO term using streams to avoid creating copies of all terms when I realized that streams are not closed automatically when collected. Those that involve underlying IO resources have to be closed as soon as possible. There's a discussion in Hibernate's issue tracker about this that is relevant.

I think we also use `File.lines` somewhere too which would run into a similar problem.

Bottom line: close your database streams!

**Comment by arteymix** [comment]
See https://docs.spring.io/spring-data/jpa/docs/current/reference/html/#repositories.query-streaming
